hs_core/hydroshare/hs_bagit.py

```diff
@@ -29,55 +27,47 @@ def delete_bag(resource):
     :param resource: the resource to delete the bag for.
     :return: none
     """
-    res_id = resource.short_id
-
-    res_path = resource.root_path
-    bagname = 'bags/{res_id}.zip'.format(res_id=res_id)
     istorage = resource.get_irods_storage()
-    if resource.resource_federation_path:
-        bagname = '{}/{}'.format(resource.resource_federation_path, bagname)
 
     # delete resource directory first to remove all generated bag-related files for the resource
-    if istorage.exists(res_path):
-        istorage.delete(res_path)
+    # TODO: Is this correct? Is it really necessary? It seems this deletes all content!
+    if istorage.exists(resource.root_path):
+        istorage.delete(resource.root_path)
 
-    # the resource bag may not exist due to on-demand bagging
-    if istorage.exists(bagname):
-        # delete the resource bag
-        istorage.delete(bagname)
+    if istorage.exists(resource.bag_path):
+        istorage.delete(resource.bag_path)
 
     # delete the bags table
+    # TODO: Why were extra bags created in the first place?
     for bag in resource.bags.all():
         bag.delete()
 
 
-def create_bag_files(resource, fed_zone_home_path=''):
+def create_bag_files(resource):
```

Comment:
@hyi @mjstealey If that is too confusing I could simply accept and ignore the file path to make the deploy work seamlessly. However -- and this is much more important -- the new semantics for files -- including no overwriting of existing files -- caused lots of tests to break. I have fixed those tests in this PR only.